tt_metal/hw/inc/dev_msgs.h

```diff
@@ -234,18 +243,42 @@ struct watcher_msg_t {
     struct debug_ring_buf_msg_t debug_ring_buf;
 };
 
+constexpr static std::uint32_t MAX_DRAM_CORES = 36;
```

Comment:
These can also go into the HAL, though are a little awkward.  Mailbox sizes can vary across architectures, so the fields can be unique per arch (that way device side code doesn't need to loop over non-existent values).  However, easiest for now to size the overall mailbox per the MAX across architectures.

Reply:
Kept it sized for MAX across all architectures for now, let me know if we want to look into changing it now